### PR TITLE
Support deduping fragments

### LINKED ISSUE #1302: Run Duplicate Marking on Fragments
The BDG-formats `Fragment` type is akin to the `org.bdgenomics.adam.models.SingleReadBucket` type, which is the core data structure in duplicate marking. If we have fragments, we should be able to run duplicate marking without the `groupBy` needed to make `SingleReadBucket`s.

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #1302 and #1303. Needs more tests and a Pandoc update. Also, I'm thinking the `reads2fragments` and `fragments2reads` CLIs should be merged down to a `transformFragments` CLI. Any disagreement?

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala [hunk 1]
```diff
@@ -484,6 +484,51 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
...
+   *   sorted.
+   */
+  private[rdd] def filesAreQuerynameSorted(filePath: String,
+                                           stringency: ValidationStringency = ValidationStringency.STRICT): Boolean = {
+    val path = new Path(filePath)
+
+    val bamFiles = getFsAndFiles(path)
+    val filteredFiles = bamFiles.filter(p => {
+      val pPath = p.getName()
+      pPath.endsWith(".bam") || pPath.endsWith(".cram") ||
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala]: could split this out into a separate method similar to `isVcfExt`

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/fragment/FragmentRDD.scala [hunk 2]
```diff
@@ -92,6 +94,16 @@ case class FragmentRDD(rdd: RDD[Fragment],
   }
 
   /**
+   * Marks reads as possible fragment duplicates.
+   *
+   * @return A new RDD where reads have the duplicate read flag set. Duplicate
+   *   reads are NOT filtered out.
+   */
+  def markDuplicates(): FragmentRDD = MarkDuplicatesInDriver.time {
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/fragment/FragmentRDD.scala]: is this saying that `MarkDuplicates` runs solely on the driver, or that the timer is?

Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/fragment/FragmentRDD.scala]: Just the timer.

**Comment by fnothaft** [comment]
Rebased. Tests and documentation are good to go now too.

**Comment by fnothaft** [comment]
I.e., this PR is good to merge.

**Comment by heuermh** [comment]
Thank you, @fnothaft!

**Pull request merged**